pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -0,0 +1,204 @@
+		if err != nil {
+			return fmt.Errorf("RegisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
+		}
+		defer conn.Close()
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+
+		infoResp, err := client.GetInfo(ctx, &registerapi.InfoRequest{})
+		if err != nil {
+			return fmt.Errorf("RegisterPlugin error -- failed to get plugin info using RPC GetInfo at socket %s, err: %v", socketPath, err)
+		}
+
+		handler, ok := pluginHandlers[infoResp.Type]
+		if !ok {
+			if err := og.notifyPlugin(client, false, fmt.Sprintf("RegisterPlugin error -- no handler registered for plugin type: %s at socket %s", infoResp.Type, socketPath)); err != nil {
+				return fmt.Errorf("RegisterPlugin error -- failed to send error at socket %s, err: %v", socketPath, err)
+			}
+			return fmt.Errorf("RegisterPlugin error -- no handler registered for plugin type: %s at socket %s", infoResp.Type, socketPath)
+		}
+
+		if infoResp.Endpoint == "" {
+			infoResp.Endpoint = socketPath
+		}
+		if err := handler.ValidatePlugin(infoResp.Name, infoResp.Endpoint, infoResp.SupportedVersions, foundInDeprecatedDir); err != nil {
+			if err = og.notifyPlugin(client, false, fmt.Sprintf("RegisterPlugin error -- plugin validation failed with err: %v", err)); err != nil {
+				return fmt.Errorf("RegisterPlugin error -- failed to send error at socket %s, err: %v", socketPath, err)
+			}
+			return fmt.Errorf("RegisterPlugin error -- pluginHandler.ValidatePluginFunc failed")
+		}
+		// We add the plugin to the actual state of world cache before calling a plugin consumer's Register handle
+		// so that if we receive a delete event during Register Plugin, we can process it as a DeRegister call.
+		actualStateOfWorldUpdater.AddPlugin(cache.PluginInfo{
+			SocketPath:           socketPath,
+			FoundInDeprecatedDir: foundInDeprecatedDir,
+			Timestamp:            timestamp,
+		})
+		if err := handler.RegisterPlugin(infoResp.Name, infoResp.Endpoint, infoResp.SupportedVersions); err != nil {
+			return og.notifyPlugin(client, false, fmt.Sprintf("RegisterPlugin error -- plugin registration failed with err: %v", err))
+		}
+
+		// Notify is called after register to guarantee that even if notify throws an error Register will always be called after validate
+		if err := og.notifyPlugin(client, true, ""); err != nil {
+			return fmt.Errorf("RegisterPlugin error -- failed to send registration status at socket %s, err: %v", socketPath, err)
+		}
+		return nil
+	}
+	return registerPluginFunc
+}
+
+func (og *operationGenerator) GenerateUnregisterPluginFunc(
+	socketPath string,
+	pluginHandlers map[string]cache.PluginHandler,
+	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
+
+	unregisterPluginFunc := func() error {
+		client, conn, err := dial(socketPath, dialTimeoutDuration)
+		if err != nil {
+			return fmt.Errorf("dial failed at socket %s, err: %v", socketPath, err)
```

Comment:
nit: Prefix error string with `UnregisterPlugin error -- dial failed...` like you did for Register already.

Ditto for the other errors in this method.